projects/optic/src/utils/checksum.ts

```diff
@@ -0,0 +1,10 @@
+import { createHash } from 'crypto';
+import stableStringify from 'json-stable-stringify';
+
+export function computeChecksum(file: any): string {
+  const hash = createHash('sha256');
+
+  hash.update(stableStringify(file));
```

Comment:
It may make more sense to just stable stringify before we checksum and then use it to create the checksum and to upload, since s3 will reject a blob that doesn't match the requested checksum.